pkg/cache/v2/simple.go

```diff
@@ -366,8 +378,65 @@ func createResponse(request *Request, resources map[string]types.ResourceWithTtl
 	}
 }
 
+// CreateDeltaWatch returns a watch for a delta xDS request which implements the Simple SnapshotCache.
 func (cache *snapshotCache) CreateDeltaWatch(request *DeltaRequest, st *stream.StreamState) (chan DeltaResponse, func()) {
-	return nil, nil
+	nodeID := cache.hash.ID(request.Node)
+	t := request.GetTypeUrl()
+
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	info, ok := cache.status[nodeID]
+	if !ok {
+		info = newStatusInfo(request.Node)
+		cache.status[nodeID] = info
+	}
+
+	// update last watch request times
+	info.mu.Lock()
+	info.lastDeltaWatchRequestTime = time.Now()
+	info.mu.Unlock()
+
+	value := make(chan DeltaResponse, 1)
+
+	// find the current cache snapshot for the provided node
+	snapshot, exists := cache.snapshots[nodeID]
+
+	// if respondDelta returns nil, there is no change in any nested resource version against the previous snapshot
```

Comment:
Does talking about the previous snapshot make sense in this context? We're just creating a watch against the current snapshot

I think this maybe leaks the fact that the incoming request might be happening in response to a snapshot change, but it requires quite a bit of mental acrobatics to get there.

Maybe just say something like `If respondDelta returns nil, there are no resources to send so we create a watch to wait for future updates`?

Reply:
Yea I see what you mean. I think it is a little hard to follow. If they want to investigate what respondDelta actually does they can go dig and should be able to figure out it just compares the resource version hashes